src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx

```diff
@@ -401,31 +409,116 @@ export default function EnvTriggerView({ filteredAppIds }: AppGroupDetailDefault
         } else {
             setFilteredWorkflows(_workflows)
             abortControllerRef.current = new AbortController()
-            updateCIMaterialList(
+            getMaterialHistory(
                 selectedCINode.id.toString(),
-                pipelineName,
-                true,
                 abortControllerRef.current.signal,
+                gitMaterialId,
+                false,
             ).catch((errors: ServerErrors) => {
-                showError(errors)
-                setErrorCode(errors.code)
+                if (!abortControllerRef.current.signal.aborted) {
+                    showError(errors)
+                }
             })
         }
     }
 
-    //NOTE: GIT MATERIAL ID
-    const refreshMaterial = (
-        ciNodeId: number,
-        pipelineName: string,
-        gitMaterialId: number,
-        abortController?: AbortController,
+    const getFilteredMaterial = async (ciNodeId: number, gitMaterialId: number, showExcluded: boolean) => {
+        const _workflows = [...filteredWorkflows].map((wf) => {
+            wf.nodes = wf.nodes.map((node) => {
+                if (node.id === ciNodeId.toString() && node.type === 'CI') {
+                    node.inputMaterialList = node.inputMaterialList.map((material) => {
+                        if (material.gitMaterialId === gitMaterialId) {
+                            material.isMaterialLoading = true
+                            material.showAllCommits = showExcluded
+                        }
+                        return material
+                    })
+                    return node
+                }
+                return node
+            })
+            return wf
+        })
+        setFilteredWorkflows(_workflows)
+        abortControllerRef.current = new AbortController()
+        getMaterialHistory(ciNodeId.toString(), abortControllerRef.current.signal, gitMaterialId, showExcluded).catch(
+            (errors: ServerErrors) => {
+                if (!abortControllerRef.current.signal.aborted) {
+                    showError(errors)
+                }
+            },
+        )
+    }
+
+    const getMaterialHistory = (
+        ciNodeId: string,
+        abortSignal: AbortSignal,
+        gitMaterialId?: number,
+        showExcluded?: boolean,
     ) => {
+        const params = {
+            pipelineId: ciNodeId,
+            materialId: gitMaterialId,
+            showExcluded: showExcluded,
+        }
+        return getCIMaterialList(params, abortSignal).then((response) => {
+            let showRegexModal = false
+            const _workflows = [...filteredWorkflows].map((wf) => {
+                wf.nodes.map((node) => {
+                    if (node.type === 'CI' && +node.id == +ciNodeId) {
```

Comment:
if both `node.id & ciNodeId` are string then can we directly do `===` here?

Reply:
done